src/integrationTest/java/uk/gov/hmcts/reform/cdapi/RetrieveCategoriesIntegrationTest.java

```diff
@@ -70,11 +83,26 @@ void shouldThrowStatusCode404ForNullCategoryId()
         assertThat((Map<String, Object>) errorResponseMap).containsEntry("http_status", HttpStatus.NOT_FOUND);
     }
 
+    @Test
+    void shouldRetrieveCategoriesWithServiceIdWithOutChildNodes()
+        throws JsonProcessingException {
+        final var response = (Categories)
+            commonDataApiClient.retrieveCaseFlagsByServiceId("HearingChannel?serviceId=BBA3"
+                                                                 + "&isChildRequired=N",
+                                                             Categories.class, path
+            );
+        assertNotNull(response);
+        assertEquals(4, response.getListOfCategory().size());
+        responseVerificationWithOutChildNodes(response);
```

Comment:
pls add a assertion to check how many expected child nodes

Reply:
Done @rajkatla-hmcts